test-framework/sudo-compliance-tests/src/sudo/flag_shell.rs

```diff
@@ -67,6 +70,39 @@ echo $0";
     Ok(())
 }
 
+#[test]
+fn shell_is_partially_canonicalized() -> Result<()> {
+    let shell_path = "/tmp/mysh";
+    let bin_link = "/tmp/bin";
+    let env = Env("ALL ALL=(ALL:ALL) NOPASSWD: /bin/sh").build()?;
+
+    Command::new("ln")
+        .args(["-s", "/bin/sh", shell_path])
+        .output(&env)?
+        .assert_success()?;
+
+    Command::new("ln")
+        .args(["-s", "/bin", bin_link])
+        .output(&env)?
+        .assert_success()?;
+
+    let output = Command::new("env")
+        .arg(format!("SHELL={shell_path}"))
+        .args(["sudo", "-s", "true"])
+        .output(&env)?;
+
+    assert!(!output.status().success());
+
+    let output = Command::new("env")
+        .arg(format!("SHELL={bin_link}/sh"))
+        .args(["sudo", "-s", "true"])
+        .output(&env)?;
+
+    assert!(output.status().success());
```

Comment:
Maybe also add a check that using a SHELL not in sudoers results in an error? Otherwise the test could pass if sudo-rs always uses /bin/sh as shell.